### PR TITLE
Temporary Poster Background

### PR DESCRIPTION
**Pull request by bagusnl**
# Main Goal
Use poster as temporary background when requested background is not yet downloaded.
New background retrieval mechanism:
1. Check if background is downloaded
2. If it's downloaded, then load the downloaded background
3. If not yet downloaded, load either last available regional background or use poster as background instead

Due to the poster having slight transparency, the switching looks a wee bit janky

This PR also contains IOException fixes especially for background loading and image cache processing (really noticeable when user uses Waifu2X)

## PR Status :
- Overall Status : Done
- Commits : Done
- Synced to base (Collapse:main) : Yes
- Build status : OK
- Crashing : No
- Bug found caused by PR : 0

### CODE REVIEW

File: CollapseLauncher/Classes/Helper/Image/ImageLoaderHelper.cs [hunk 1]
```diff
@@ -276,7 +275,7 @@ private static async void LoadImageCropperDetached(string filePath, ImageCropper
 
             dialogOverlay!.IsPrimaryButtonEnabled = true;
         }
-
+
```
Comment (Cryotechnic) [inline CollapseLauncher/Classes/Helper/Image/ImageLoaderHelper.cs:278]: Remove whitespace

File: CollapseLauncher/XAMLs/MainApp/Pages/OOBE/OOBESelectGameBG.xaml.cs [hunk 2]
```diff
@@ -90,9 +91,18 @@ internal static async Task<bool> TryLoadGameDetails(PresetConfig config = null)
...
+                                    GameNameType.Honkai => Path.Combine(AppFolder,   @"Assets\Images\GamePoster\poster_honkai.png"),
+                                    GameNameType.Genshin => Path.Combine(AppFolder,  @"Assets\Images\GamePoster\poster_genshin.png"),
+                                    GameNameType.StarRail => Path.Combine(AppFolder, @"Assets\Images\GamePoster\poster_starrail.png"),
+                                    GameNameType.Zenless => Path.Combine(AppFolder,  @"Assets\Images\GamePoster\poster_zzz.png"),
+                                    _ => AppDefaultBG
+                                };
 
                 // TODO: Use FallbackCDNUtil to get the sprites
-                _gamePosterPath = await ImageLoaderHelper.GetCachedSpritesAsync(FallbackCDNUtil.TryGetAbsoluteToRelativeCDNURL(config.ZonePosterURL, "metadata/"), default);
+                //_gamePosterPath = await ImageLoaderHelper.GetCachedSpritesAsync(FallbackCDNUtil.TryGetAbsoluteToRelativeCDNURL(config.ZonePosterURL, "metadata/"), default);
```
Comment (Cryotechnic) [inline CollapseLauncher/XAMLs/MainApp/Pages/OOBE/OOBESelectGameBG.xaml.cs:105]: Remove if not in use, or uncomment if you commented it for testing

**Review by Cryotechnic — CHANGES_REQUESTED** [review]
Is the commented code intentional? Otherwise, looks good

File: CollapseLauncher/XAMLs/MainApp/Pages/OOBE/OOBESelectGameBG.xaml.cs (hunk 2 above)
Comment (bagusnl) [reply CollapseLauncher/XAMLs/MainApp/Pages/OOBE/OOBESelectGameBG.xaml.cs:105]: Intentional as the code might be used again if the local loading method is removed or we want to use different posters for OOBE

File: CollapseLauncher/Classes/Helper/Image/ImageLoaderHelper.cs (hunk 1 above)
Comment (bagusnl) [reply CollapseLauncher/Classes/Helper/Image/ImageLoaderHelper.cs]: fixed

**Comment by bagusnl** [comment]
Merging as no more review came :sadge:

**Pull request merged**